uint8[] data
string dest
---
bool successful